core/src/main/java/org/hisp/dhis/android/core/arch/d2/internal/JavaDIClasses.kt

```diff
@@ -47,7 +47,7 @@ internal val javaDIClasses = module {
     single { FileResourceCleaner(get(), get(), get()) }
     single { NoteUniquenessManager(get()) }
     single { PeriodHelper(get(), get(), get(), get(), get()) }
-    single { PeriodParser(get()) }
+    single { PeriodParser() }
```

Comment:
This entry could be removed if the class PeriodParser is annotated with @Singleton. It is not possible to annotate Java classes, that is why it was here.
Anyway, it can be replaced in separate task to refactor all these classes to kotlin